Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs

```diff
@@ -808,6 +804,10 @@ private bool CheckForCollisionBetween()
             //Order cells by distance from the last position
             foreach (var cell in cells)
             {
+                if (BlockerRegistry.CheckForCollisionBetweenCallback(this, LastPos, ExactPosition.ToIntVec3() == cell ? ExactPosition : cell.ToVector3Shifted()))
```

Comment:
The `CheckForCollisionBetweenCallback` is a low cost check, to significantly improve performance vs checking every cell.  It was added to allow radial shields to skip the per-cell checking on line 811.  It needs to stay outside the loop (or it should just be removed, since `CheckCellForCollision` serves the inside-the-loop cases).

That said, it also needs to not prevent the projectile from checking the cells *up to* the collision point still need to be checked.  Unfortunately, this will require a bit of refactoring.

1. `CheckForCollisionBetweenCallback` probably needs to return the cell where the collision happens.  
2. It needs to not immediately call the projectile cleanup/impact code.
3. If it returns a cell, the `foreach` on 805 needs to break when it reaches that cell.
4. If the foreach `reaches` that cell, the projectile cleanup/impact needs to run.  This could be a separate callback, or by having `CheckForCollisionBetweenCallback` also provide a continuation to call if nothing intercepts the projectile first.

Reply:
Outdated now